src/app.controller.ts

```diff
@@ -0,0 +1,39 @@
+import { Controller } from "@nestjs/common";
+import { GrpcMethod } from "@nestjs/microservices";
+import {
+	GetAffluenceRequestDto,
+	GetClientsCountRequestDto,
+	GetPromotionCheckoutsCountRequestDto
+} from "./analytics.dto";
+import { AppService } from "./app.service";
+import { ANALYTICS_SERVICE_NAME, GetAffluenceResponse, GetPromotionsRankingRequest } from "./general.pb";
+
+@Controller("analytics")
+export class AppController {
+	constructor(private readonly appService: AppService) {}
+
+	@GrpcMethod(ANALYTICS_SERVICE_NAME, "getAffluence")
+	public affluence(payload: GetAffluenceRequestDto): Promise<GetAffluenceResponse> {
+		return this.appService.getAffluence(payload.shopId, payload.startDate, payload.endDate);
```

Comment:
```suggestion
		return this.appService.getAffluence(payload);
```
Tu peux directement passer payload

Reply:
done